2019-10-26  Frank Siyuan Xing  <frankx@lnxsrv07.seas.ucla.edu>

        * timespec.h (timespec_cmp): The timespec_cmp function has been
        modified. The function now returns 1 if the time a is later than time
        b. It returns -1 if the time b is later than time a. It returns 0 if
        time a is equal to time b.


--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2019-10-26 11:15:52.854156000 -0700
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec) {
+    return 1;
+  }
+  else if (a.tv_sec < b.tv_sec) {
+    return -1;
+  }
+  else {
+    return 0;
+  }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
